test/test_nested_tensor_masking.py

```diff
@@ -3,42 +3,257 @@
+        # mask_dim = None
+        res_tensor, res_mask = original_nt.to_tensor_mask()
+        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
+                                   [[[[5, 6], [7, 8]]]]])
+        exp_mask = torch.tensor(True)
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+
+        # mask_dim = 0
+        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=0)
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+
+        # mask_dim = 1
+        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=1)
+        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
+                                   [[[[5, 6], [7, 8]]]]])
+        exp_mask = torch.tensor([True, True])
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+
+        # mask_dim = 2
+        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=2)
+        exp_mask = torch.tensor([[True], [True]])
+        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
+                                   [[[[5, 6], [7, 8]]]]])
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)
+        
+        # mask_dim = 3
+        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=3)
+        exp_mask = torch.tensor([[[True]], [[True]]])
+        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
+                                   [[[[5, 6], [7, 8]]]]])
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor)    
+
+        # mask_dim = 4
+        res_tensor, res_mask = original_nt.to_tensor_mask(mask_dim=4)
+        exp_mask = torch.tensor([[[[True, True]]],[[[True, True]]]])
+        exp_tensor = torch.tensor([[[[[1, 2], [3, 4]]]],
+                                   [[[[5, 6], [7, 8]]]]])
+        self.assertEqual(res_mask, exp_mask)
+        self.assertEqual(res_tensor, exp_tensor) 
+
+    #
+    # Group of tests to test nested_tensor_from_tensor_mask() 
+    #
+    def test_nested_tensor_from_tensor_mask_dim_none(self):
+        tensor, mask = original_nt.to_tensor_mask()
+        self.assertRaisesRegex(RuntimeError, 
+                               "Mask has to have dimention > 0", 
+                               lambda: nt.nested_tensor_from_tensor_mask(tensor, mask))
+
+    def test_nested_tensor_from_tensor_mask_dim_1(self):
+        tensor, mask = original_nt.to_tensor_mask(mask_dim = 1)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask)
+        self.assertEqual(expected_nt0, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=0)
+        self.assertEqual(expected_nt0, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=1)
+        self.assertEqual(expected_nt1, res_nt)
+
+        res_nt = nt.nested_tensor_from_tensor_mask(tensor, mask, nested_dim=2)
```

Comment:
Two nestedtensors with different nested_dim shouldn't be considered equal. Why does the result form this match that of all the other ones above?

Reply:
It's probably correct to raise a ValueError if the passed mask doesn't allow the reconstruction of a NestedTensor with the specified nested_dim

Reply:
Updated